sdk/core/Azure.Core/src/ResponseExceptionExtensions.cs

```diff
@@ -35,33 +36,39 @@ public static RequestFailedException CreateRequestFailedException(this Response
 
         public static async Task<RequestFailedException> CreateRequestFailedExceptionAsync(string message, Response response, bool async)
         {
-            StringBuilder detailsBuilder = new StringBuilder()
+            StringBuilder messageBuilder = new StringBuilder()
+                .AppendLine(message)
                 .Append("Status: ")
-                .AppendLine(response.Status.ToString())
-                .Append("ReasonPhrase: ")
-                .AppendLine(response.ReasonPhrase)
-                .AppendLine()
-                .AppendLine("Headers:");
-            foreach (var responseHeader in response.Headers)
-            {
-                detailsBuilder.AppendLine($"{responseHeader.Name}: {responseHeader.Value}");
-            }
+                .Append(response.Status.ToString())
+                .Append(" (")
+                .Append(response.ReasonPhrase)
+                .AppendLine(")");
 
             if (response.ContentStream != null &&
-                response.Headers.ContentType?.StartsWith("text/", StringComparison.OrdinalIgnoreCase) == true)
+                ResponseClassifier.IsTextResponse(response, out var encoding))
             {
-                detailsBuilder.AppendLine()
+                messageBuilder
+                    .AppendLine()
                     .AppendLine("Content:");
 
-                using (var streamReader = new StreamReader(response.ContentStream))
+                using (var streamReader = new StreamReader(response.ContentStream, encoding))
                 {
                     string content = async ? await streamReader.ReadToEndAsync() : streamReader.ReadToEnd();
 
-                    detailsBuilder.AppendLine(content);
+                    messageBuilder.AppendLine(content);
                 }
             }
 
-            return new RequestFailedException(response.Status, message, detailsBuilder.ToString());
+
+            messageBuilder
+                .AppendLine()
+                .AppendLine("Headers:");
+            foreach (var responseHeader in response.Headers)
```

Comment:
This makes me nervous, given that sensitive information may appear in headers.   II realize that this is more likely in request headers where auth-releated things are passed, but I'd like to suggest that we have a white list for headers to include (or at least a black list to prevent known sensitive things.)

Reply:
What would you put in the list for responses?

Reply:
I don't want to block work on redaction while we don't have clear guidance around it. https://github.com/Azure/azure-sdk-for-net/issues/6156 is filed to track adding support for both log and exception redaction.

Reply:
Personally, I'd explicitly filter `Authorization` out.   While I wouldn't expect to see that ever happen in a response, and would certainly consider it a poor design, I've seen services echo the full set of request headers with a failure response and not sanitize.  _(sadly, one of these was a popular credit card processor who also sent the card information back in the request body with a pass/fail flag for validation and claimed everything was fine because SSL)_.   Even if the Azure services are well-behaved and never do this, the cost is negligible.

Outside of that, I'd white-list the standard set of response headers that would be interesting/helpful and cut the rest out to eliminate noise.    Likewise with custom headers; I'd white-list those that the service is known to return and which may be useful and ignore the rest. 

For the standard ones, I'd suggest:

- Age
- Allow
- Cache-Control
- Content-Disposition
- Content-Encoding
- Content-Language
- Content-Length
- Content-MD5
- Content-Range
- Content-Type
- Date
- ETag
- Expires
- Last-Modified
- Link
- Location
- Proxy-Authenticate
- Retry-After
- Server
- Strict-Transport-Security
- Transfer-Encoding
- Warning